From 3a648d11a3b9c7e19affd47edd05ce3713de9add Mon Sep 17 00:00:00 2001
From: Nick Wellnhofer <wellnhofer@aevum.de>
Date: Thu, 25 Jul 2024 17:07:18 +0200
Subject: [PATCH] parser: Make xmlParseChunk return an error if parser was
 stopped

This regressed after enhancing the disableSAX member in 2.13.

Should fix #777.
---
 parser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parser.c b/parser.c
index 10695ecfd..365025c5e 100644
--- a/parser.c
+++ b/parser.c
@@ -11575,7 +11575,7 @@ xmlParseChunk(xmlParserCtxtPtr ctxt, const char *chunk, int size,
         xmlHaltParser(ctxt);
     }
 
-    if ((ctxt->errNo != XML_ERR_OK) && (ctxt->disableSAX == 1))
+    if ((ctxt->errNo != XML_ERR_OK) && (ctxt->disableSAX != 0))
         return(ctxt->errNo);
 
     if ((end_in_lf == 1) && (ctxt->input != NULL) &&
-- 
GitLab

